tools/data_manipulation/xarray/xarray_metadata_info.xml

```diff
@@ -1,41 +1,28 @@
-<tool id="netcdf_metadata_info" name="Netcdf Metadata Info" version="0.1.0">
+<tool id="xarray_metadata_info" name="Netcdf Metadata Info" version="0.15.1">
     <description>summarize content of a Netcdf file</description>
     <requirements>
-        <requirement type="package" version="1.1.6">netcdf-metadata-info</requirement> 
+        <requirement type="package" version="3">python</requirement>
+        <requirement type="package" version="1.5.3">netcdf4</requirement>
+        <requirement type="package" version="0.15.1">xarray</requirement>
+        <requirement type="package" version="0.7.0">geopandas</requirement>
+        <requirement type="package" version="1.7.0">shapely</requirement> 
     </requirements>
     <command detect_errors="exit_code"><![CDATA[
-        netcdf-metadata-info '$input'
-        &&
-        while read -r l;do
-            a=\$(echo \$l | cut -d' ' -f1);echo \$l>dimensions_\$a
-        ;done <variables.tabular
-        &&
-        rm dimensions_VariableName 
-        &&
-        for f in dimensions_*; do cat \$f | sed 's/ /\t\n/g' | sed '\$s/$/ /' >\$f.tabular ; done
-        &&
-        for f in dimensions_*.tabular;do
-            awk 'NR % 2 != 0' \$f > \$f.2
-            &&
-            sed 1d \$f.2 > \$f 
-            &&
-            rm \$f.2
-        ;done
-        &&
-        ncdump -h '$input' > '$info'
+        python3 '$__tool_directory__/xarray_tool.py' '$input' --info '$info' --summary '$output'
     ]]></command>
     <inputs>
         <param type="data" name="input" label="Netcdf file" format="netcdf,h5" help="Netcdf file you need information about."/>
     </inputs>
     <outputs>
-        <data name="output" format="tabular" label="Metadata infos from ${input.name}.Variables.tab" from_work_dir="variables.tabular"/>
+        <data name="output" format="tabular" label="Metadata infos from ${input.name}" from_work_dir="variables.tabular"/>
         <data name="info" label="info file" format="txt"/>
     </outputs>
     <tests>
         <test>
              <param name="input" value="dataset-ibi-reanalysis-bio-005-003-monthly-regulargrid_1510914389133.nc"/>
              <param name="output" value="Metadata_infos_from_dataset-ibi-reanalysis-bio-005-003-monthly-regulargrid_1510914389133.nc.Variables.tab"/>
-             <param name="info" value="info_file.txt"/>
+             <output name="info" value="info_file.txt" compare="sim_size" delta="10"/>
```

Comment:
maybe we can use asserts for testing?

Reply:
Yes. we should use something a bit more robust... Ideally, we should try to extract some information from the two files and check they are correct.